### PR TITLE
Add doxygen info for DynElimination pass

### PR DESCRIPTION
**Pull request by yimeisun123**

### CODE REVIEW

**Comment by yimeisun123** [comment]
Here is the outlook for the generated description.
dyn_elim

File: src/ngraph/pass/dyn_elimination.hpp [hunk 1]
```diff
@@ -23,6 +23,33 @@ namespace ngraph
 {
     namespace pass
     {
+        /// \brief The DynElimination pass finds dynamic operations in a graph whose
+        /// shape relevant inputs have already been resolved to static values, and
+        /// replaces those dynamic operations with the equivalent operations using
+        /// statical inputs and attributes.
```
Comment (rkimballn1) [inline src/ngraph/pass/dyn_elimination.hpp:29]:
```suggestion
        /// static inputs and attributes.
```

File: src/ngraph/pass/dyn_elimination.hpp [hunk 2]
```diff
@@ -23,6 +23,33 @@ namespace ngraph
...
+        /// \details This pass should be executed after the ConstantFolding pass.
+        ///
+        /// The ConstantFolding and DynElimination pass are used together to transform
```
Comment (rkimballn1) [inline src/ngraph/pass/dyn_elimination.hpp:32]:
```suggestion
        /// The ConstantFolding and DynElimination passes are used together to transform
```

File: src/ngraph/pass/dyn_elimination.hpp [hunk 3]
```diff
@@ -23,6 +23,33 @@ namespace ngraph
...
+        /// dynamic operations in a computation graph to static operations when the
+        /// graph is executed with input data.
+        ///
+        /// In the example shown below, the original graph is constructed with dynamic
+        /// broadcast operation. When the graph is executed with input data, the input
+        /// shapes become available, by applying the ConstantFolding and DynElimination
+        /// pass, the graph is updated with dynamic broadcast being replaced by static
```
Comment (rkimballn1) [inline src/ngraph/pass/dyn_elimination.hpp:39]:
```suggestion
        /// pass, the graph is updated with dynamic broadcast being replaced by a static
```

**Comment by yimeisun123** [comment]
@rkimballn1 🙏 for the update.

File: src/ngraph/pass/dyn_elimination.hpp [hunk 4]
```diff
@@ -23,6 +23,33 @@ namespace ngraph
...
+        ///
+        /// In the example shown below, the original graph is constructed with dynamic
+        /// broadcast operation. When the graph is executed with input data, the input
+        /// shapes become available, by applying the ConstantFolding and DynElimination
+        /// pass, the graph is updated with dynamic broadcast being replaced by a static
+        /// broadcast operation.
+        /// <table>
+        /// <tr>
+        ///     <th>Original</th>
+        ///     <th>After %ConstantFolding
```
Comment (gaurides) [inline src/ngraph/pass/dyn_elimination.hpp:44]:
```suggestion
        ///     <th>After %ConstantFolding</th>
```

**Pull request merged**